### ISSUE TITLE
'markdownlint: Lint all Markdown files in the workspace' ignores `markdownlint.configFile`

### ISSUE THREAD

**Issue by jack5github** [DavidAnson/vscode-markdownlint#461]
**What did you do?**
I used 'markdownlint: Lint all Markdown files in the workspace' from the Command Palette in a workspace with a `.vscode/settings.json` file that contains `"markdownlint.configFile": ".vscode/.markdownlint-cli2.yaml"`. `.vscode/.markdownlint-cli2.yaml` is a YAML configuration as specified below, which works when used to edit individual files using the VS Code editor.
```yaml
config:
  MD013: false
  MD022:
    lines_below: 0
  MD024: false
  MD028: false
  MD029: false
  MD031: false
  MD032: false
  MD034: false
  MD037: false
  MD040: false
  MD049:
    style: underscore
  MD050:
    style: asterisk
  MD058: false
  MD060:
    style: compact
```

**What did you expect to happen?**
Linting should obey the configuration specified in `.vscode/.markdownlint-cli2.yaml`.

**What actually happened?**
Linting instead obeys default markdownlint settings. If the file is open when this linting happens, these incorrect errors will appear as VS Code problems. Regardless, these incorrect problems are visible in VS Code. Re-opening the file clears the incorrect problems.

**What messages or errors were there?**
Sensitive information is omitted with '...'.
```
Linting workspace folder "file:///c%3A/Users/.../Documents/Visual%20Studio%20Code/..."...
markdownlint-cli2 v0.23.2 (markdownlint v0.41.1)
Finding: **/*.{md,mkd,mdwn,mdown,markdown,markdn,mdtxt,mdtext,workbook} !**/*.code-search !**/bower_components !**/node_modules !**/.git !**/vendor
Linting: 6 files
Summary: ... issues in ... files
....md:23:12 error MD049/emphasis-style Emphasis style [Expected: asterisk; Actual: underscore]
```

**How can the issue be reproduced?**
1. Create `.vscode/.markdownlint-cli2.yaml` as shown above.
2. Add `"markdownlint.configFile": ".vscode/.markdownlint-cli2.yaml"` to `.vscode/settings.json`.
3. Create several Markdown files that violate the specified rules.
4. Restart VS Code to ensure configuration is loaded.
5. Run 'markdownlint: Lint all Markdown files in the workspace' from the Command Palette.

**What version were you using?**
- VS Code: 1.135.0
- markdownlint extension: 0.62.1
- Node.js: 25.2.1
- markdownlint-micromark: 0.1.2
- markdownlint-rule-helpers: 0.21.0

**What operating system were you using?**
Microsoft Windows 11 Pro 10.0.26200
